src/services/array-functions.js

```diff
@@ -1,29 +1,46 @@
 
 //in the function map, create a new array and store in a variable
-//loop theArray and call the fnc for each thing in the array, 
+//loop theArray and call the fnc for each thing in the array,
 //     passing in the item from the current loop into the call to fnc
 //add the returned value from fnc to the new array
 //return the new array
 export function map(theArray, fnc){
-
+  let newArray = [];
+  for (let i = 0; i < theArray.length; i++){
+    newArray.push(fnc(theArray[i]));
+  }
+  return newArray;
 }
 
 //create a new array
-//loop theArray and call the fnc for each thing in the array, 
+//loop theArray and call the fnc for each thing in the array,
 //     passing in the item from the current loop
 //fnc will return true or false, if true add the item to the new array else do not
 //return the new array
 export function filter(theArray, fnc){
-
+  let newArray = [];
+  for (let i = 0; i < theArray.length; i++){
+    // console.log(theArray[i]);
+    if (fnc(theArray[i])){
+      newArray.push(theArray[i]);
+    };
+  };
+  return newArray;
 }
 
 
-//loop theArray and call the fnc for each thing in the array, 
+//loop theArray and call the fnc for each thing in the array,
 //     passing in the item from the current loop
-//fnc will return true or false, if true return the item 
+//fnc will return true or false, if true return the item
 //return null
 export function find(theArray, fnc){
-
+  let newArray= [];
+  for (let i = 0; i < theArray.length; i++){
+    if(fnc(theArray[i])){
+      newArray.push(theArray[i]);
```

Comment:
need to think about the find function some more. Find is suppose to return one single item from theArray. 
we do not need a newArray and we don' need to push to it
your if statement is good, but just needs to return the item from the array